### PR TITLE
V11.x.x modal

### PR DESCRIPTION
**Pull request by jesstelford**

### CODE REVIEW

File: polaris-react/src/components/Modal/Modal.stories.tsx [hunk 1]
```diff
@@ -563,3 +563,90 @@ export function WithLongContentNoScroll() {
     </Modal>
   );
 }
+
```
Comment (alex-page) [inline polaris-react/src/components/Modal/Modal.stories.tsx:566]: Thanks for adding all these missing examples!

File: polaris-react/src/components/Modal/components/CloseButton/CloseButton.scss [hunk 2]
```diff
@@ -31,4 +31,13 @@
   &.titleHidden {
     margin: var(--p-space-2);
   }
+
+  #{$se23} & {
+    margin-top: calc(-1 * var(--p-space-1));
+    margin-bottom: calc(-1 * var(--p-space-1));
```
Comment (alex-page) [inline polaris-react/src/components/Modal/components/CloseButton/CloseButton.scss:37]: I think this was done to ensure we still have a large clickable area on the close button. Looks good to me.

File: polaris-react/src/components/Modal/components/Dialog/Dialog.scss [hunk 3]
```diff
@@ -94,6 +94,12 @@ $large-width: 980px;
       height: unset;
     }
   }
+
+  #{$se23} & {
+    /* Setting background color on the header causes it to show through the
+       border radius, so we have to hide the overflow to maintain the radius */
+    overflow: hidden;
+  }
```
Comment (alex-page) [inline polaris-react/src/components/Modal/components/Dialog/Dialog.scss:102]: We should make sure to test this for edge cases. Makes sense to me though.

**Review by alex-page — APPROVED** [review]
